client-course-schedulizer/src/components/reuseables/AddNonTeachingLoadPopover/AddNonTeachingLoadPopover.tsx

```diff
@@ -0,0 +1,54 @@
+import { yupResolver } from "@hookform/resolvers/yup";
+import { Box, Button, Grid, Typography } from "@material-ui/core";
+import { GridItemCheckboxGroup, GridItemTextField } from "components";
+import React from "react";
+import { FormProvider, useForm } from "react-hook-form";
+import {
+  addNonTeachingLoadSchema,
+  NonTeachingLoadInput,
+  Term,
+  useAddSectionToSchedule,
+} from "utilities";
+
+const SPACING = 2;
+
+export const AddNonTeachingLoadPopover = () => {
+  const { addNonTeachingLoadToSchedule } = useAddSectionToSchedule();
+
+  const onSubmit = () => {
+    return (data: NonTeachingLoadInput) => {
+      addNonTeachingLoadToSchedule(data);
+    };
+  };
+
+  const methods = useForm<NonTeachingLoadInput>({
+    criteriaMode: "all",
+    resolver: yupResolver(addNonTeachingLoadSchema),
+  });
+  return (
+    <FormProvider {...methods}>
+      <form className="popover-container">
+        <Box>
+          <Typography className="popover-title" variant="h4">
+            Add Non-Teaching Load
+          </Typography>
+        </Box>
+        <Grid container spacing={SPACING}>
+          <GridItemTextField label="Name" />
```

Comment:
Is this wording better?

```suggestion
          <GridItemTextField label="Activity" />
```